crates/vm/levm/src/vm.rs

```
@@ -1303,6 +1371,226 @@ impl VM {
+        // TODO: avoid clone()
+        for auth_tuple in self.authorization_list.clone().unwrap_or_default() {
+            let chain_id_not_equals_this_chain_id = auth_tuple.chain_id != self.env.chain_id;
+            let chain_id_not_zero = !auth_tuple.chain_id.is_zero();
+
+            // 1. Verify the chain id is either 0 or the chain’s current ID.
+            if chain_id_not_zero && chain_id_not_equals_this_chain_id {
+                continue;
+            }
+
+            // 2. Verify the nonce is less than 2**64 - 1.
+            // NOTE: nonce is a u64, it's always less or equal to u64::MAX
+            if auth_tuple.nonce == u64::MAX {
+                continue;
+            }
+
+            // 3. authority = ecrecover(keccak(MAGIC || rlp([chain_id, address, nonce])), y_parity, r, s)
+            //      s value must be less than or equal to secp256k1n/2, as specified in EIP-2.
+            let order_bytes = Secp256k1::ORDER.to_be_bytes();
+            let n = U256::from_big_endian(&order_bytes);
+            let n_over_2 = n.checked_div(U256::from(2)).ok_or(VMError::Internal(
+                InternalError::ArithmeticOperationOverflow,
+            ))?;
+
+            if auth_tuple.s_signature > n_over_2 || U256::zero() >= auth_tuple.s_signature {
+                continue;
+            }
+            if auth_tuple.r_signature > n || U256::zero() >= auth_tuple.r_signature {
+                continue;
+            }
+            if auth_tuple.v != U256::one() && auth_tuple.v != U256::zero() {
+                continue;
+            }
+
+            let rlp_buf =
+                (auth_tuple.chain_id, auth_tuple.address, auth_tuple.nonce).encode_to_vec();
+
+            let mut hasher = Keccak256::new();
+            hasher.update([MAGIC]);
+            hasher.update(rlp_buf);
+            let bytes = &mut hasher.finalize();
+
+            let Ok(message) = Message::parse_slice(bytes) else {
+                continue;
+            };
```

Are we losing relevant information by discarding the errors by using this pattern (`let Ok(...) else { continue;}`)?